### PR TITLE
Admin: Enhancing user experience in repository creation

### PR DESCRIPTION
**Pull request by rldhont**
In the admin lizmap repository creation page, the form has been enhanced with:

* help strings
* new organization fo fields with path as first, label as second and id as third
* autocompletion for label and id based on selected path

lizmap-update-config-section-form

Funded by 3liz https://3liz.com

### CODE REVIEW

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 1]
```diff
@@ -133,8 +133,11 @@ form.admin_services.message.data.saved=Data has been saved.
 form.admin_section.h1.create=Create a repository
 form.admin_section.h1.modify=Modify the repository
 form.admin_section.repository.label=Id
+form.admin_section.repository.help=Repository id for machine, any characters between a-z and 0-9
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:136]:
```suggestion
form.admin_section.repository.help=Repository unique identifier, must only contain characters between a-z and 0-9 (no space, no special character)
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 2]
```diff
@@ -133,8 +133,11 @@ form.admin_services.message.data.saved=Data has been saved.
...
 form.admin_section.repository.label.label=Label
+form.admin_section.repository.label.help=Repository title for human, it will display in the interface
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:138]:
```suggestion
form.admin_section.repository.label.help=Repository label, which will be displayed in the interface
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 3]
```diff
@@ -133,8 +133,11 @@ form.admin_services.message.data.saved=Data has been saved.
...
 form.admin_section.repository.path.label=Local folder path
+form.admin_section.repository.path.help=The directory in which QGIS Projects and lizmap config will be displayed
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:140]:
```suggestion
form.admin_section.repository.path.help=The directory in which QGIS Projects and Lizmap configuration file must be added
```

File: lizmap/www/assets/js/admin/config_section.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+window.onload = () => {
+    const repIdInput = document.getElementById('jforms_admin_config_section_repository');
+    console.log(repIdInput.hasAttribute('readonly'));
```
Comment (mdouchin) [inline lizmap/www/assets/js/admin/config_section.js:3]: Remove `console.log` ?

**Review by mdouchin — CHANGES_REQUESTED** [review]
Some propositions :D

**Comment by rldhont** [comment]
Thnaks @mdouchin all has been resolved

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 5]
```diff
@@ -133,8 +133,11 @@ form.admin_services.message.data.saved=Data has been saved.
 form.admin_section.h1.create=Create a repository
 form.admin_section.h1.modify=Modify the repository
 form.admin_section.repository.label=Id
+form.admin_section.repository.help=Repository unique identifier, must only contain characters between a-z and 0-9 (no space, no special character)
```
Comment (Gustry) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:136]:
```suggestion
form.admin_section.repository.help=Repository unique identifier, must only contain characters between a-z and 0-9 (no space, no special character). This identifier can not be edited later.
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 6]
```diff
@@ -133,8 +133,11 @@ form.admin_services.message.data.saved=Data has been saved.
...
 form.admin_section.repository.label.label=Label
+form.admin_section.repository.label.help=Repository label, which will be displayed in the interface
 form.admin_section.repository.path.label=Local folder path
+form.admin_section.repository.path.help=The directory in which QGIS Projects and Lizmap configuration file must be added
```
Comment (Gustry) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:140]:
```suggestion
form.admin_section.repository.path.help=The directory in which QGIS projects and Lizmap configuration file must be stored.
```

**Review by Gustry — APPROVED** [review]

**Comment by rldhont** [comment]
@mdouchin locale has been updated

**Review by nboisteault — APPROVED** [review]

**Review by mdouchin — APPROVED** [review]

**Pull request merged**